semantics/src/traversal/expressions.rs

```diff
@@ -263,6 +263,29 @@ fn expr_bin_operation(
     unreachable!()
 }
 
+fn expr_unary_operation(
+    scope: Shared<BlockScope>,
+    context: Shared<Context>,
+    exp: &Spanned<fe::Expr>,
+) -> Result<ExpressionAttributes, SemanticError> {
+    if let fe::Expr::UnaryOperation { op, operand } = &exp.node {
+        if let fe::UnaryOperator::USub = &op.node {
+            let _operand_attributes =
+                expr_with_value_move(Rc::clone(&scope), Rc::clone(&context), operand)?;
+            // No matter what type the operand was before, the minus symbol turns it into an
+            // I256 just like all positive values default to U256.
+            return Ok(ExpressionAttributes::new(
```

Comment:
I just want to highlight this part. As far as I understand this means that any literal such as `-1` will be treated as an `I256`  which seems inline with the fact that any positive number such as `1` defaults to `U256`. But it also means that expressions such as `-x` will default to `I256`  even if `x` was something like an `U8` before.

Reply:
Oh and I should probably reject using `-` on any operand that isn't of type `Numeric`?

Reply:
> But it also means that expressions such as -x will default to I256 even if x was something like an U8 before.

We could also make sure that `x` is a literal. Do we want to have expressions like `- (foo + 42)`?

> Oh and I should probably reject using - on any operand that isn't of type Numeric?

:+1:

Reply:
>Do we want to have expressions like `- (foo + 42)`?

It's legal in Rust and Python and I think disallowing it might make the language more awkward to use so I'd say we should allow it.